eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ServiceClientInstantiationCheck.java

```diff
@@ -54,40 +99,39 @@
     public void beginTree(DetailAST root) {
         hasServiceClientAnnotation = false;
         isAsync = false;
-        isImplPackage = false;
     }
 
     @Override
     public void visitToken(DetailAST token) {
-        if (isImplPackage) {
-            return;
-        }
-
         switch (token.getType()) {
-            case TokenTypes.PACKAGE_DEF:
-                String packageName = FullIdent.createFullIdent(token.findFirstToken(TokenTypes.DOT)).getText();
-                isImplPackage = packageName.contains(".implementation");
+            case TokenTypes.IMPORT:
+                addImportedClassPath(token);
                 break;
             case TokenTypes.CLASS_DEF:
                 hasServiceClientAnnotation = hasServiceClientAnnotation(token);
-                if (hasServiceClientAnnotation) {
-                    checkServiceClientNaming(token);
+                if (!hasServiceClientAnnotation) {
```

Comment:
If the class does not have `@ServiceClient` annotation, is it possible to terminate this custom checkstyle evaluation i.e. no more calls to `visitToken()`? If that's possible, you could simplify the other switch cases by not having to check `if (!hasServiceClientAnnotation)`

Reply:
I found no way to do an earlier termination of tree traversal,